sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,489 @@ def reduce_inequalities(inequalities, symbols=[]):
+    >>> from sympy.abc import x, y, z
+    >>> from sympy import Eq
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> A, B, C, variables, constraints = _linear_programming_to_matrix([r1, r2, r3], x + y + 5*z)
+    >>> A
+    Matrix([
+    [ 0, 1,  2],
+    [-1, 0, -3],
+    [ 2, 1,  7]])
+    >>> B
+    Matrix([
+    [ 3],
+    [-2],
+    [ 5]])
+    >>> C
+    Matrix([[1, 1, 5]])
+    >>> variables
+    [x, y, z]
+    >>> constraints
+    [y + 2*z <= 3, -x - 3*z <= -2, 2*x + y + 7*z <= 5]
+    >>> A, B, C, variables, constraints = _linear_programming_to_matrix([Eq(x, 3)], x*10)  # x = 3 become x >= 3 and x <= 3
+    >>> A
+    Matrix([
+    [ 1],
+    [-1]])
+    >>> B
+    Matrix([
+    [ 3],
+    [-3]])
+    >>> C
+    Matrix([[10]])
+    >>> variables
+    [x]
+    >>> constraints
+    [x <= 3, -x <= -3]
+    """
+    variables = objective.free_symbols
+    standard_constraints = []
+    eqns = []
+
+    for rel in constraints:
+        if not isinstance(rel, (Relational, AppliedBinaryRelation)):
+            raise TypeError(f"{rel} is not relational.")
+        if type(rel) in [Lt, Gt] or isinstance(rel, AppliedBinaryRelation) and rel.function in [Q.lt, Q.gt]:
+            raise TypeError("Strict inequalities are not allowed in linear programming.")
+        if type(rel) == Ne or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ne:
+            raise TypeError("'not equal to' is not allowed in linear programming.")
+
+        variables = variables | rel.free_symbols
+
+        if type(rel) == Le or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.le:
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+        elif type(rel) == Ge or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ge:
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        elif type(rel) == Eq or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.eq:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        else:
+            raise TypeError(f"Unrecognized relation: {rel}")
+
+    variables = sorted(variables, key=lambda v: str(v)) # order symbols lexicographically
+
+    A, B = linear_eq_to_matrix(eqns, *variables)
+    C = linear_eq_to_matrix(objective, *variables)[0] # constant terms can be safely ignored here
+
+    return A, B, C, variables, standard_constraints
+
+def linprog_from_equations(constraints, objective):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method. All coefficients must be Rational or
+    floats.
```

Comment:
The name of this function should indicate that it maximises the objective function.

Reply:
I guess I could name it `linprog_maximize_from_equations`. My reasoning for naming it this is that I think it's more important to indicate that it does the same thing as `linprog_from_matrices`. Also, it can be used for minimization problems if you just make the objective function negative.